src/main/java/com/fireball1725/defaultworldgenerator/gui/GuiCreateCustomWorld.java

```diff
@@ -57,40 +49,33 @@ public void initGui() {
                                     Log.fatal(blockName + " not found, please check the config file...");
                                 } else {
 
-                                    flatworldconfig += blockQty + "x" + blockID;
+                                    flatworldconfig.append(blockQty)
+                                                   .append("x")
+                                                   .append(blockID);
                                     if (blockMeta != 0) {
-                                        flatworldconfig += ":" + blockMeta;
+                                        flatworldconfig.append(":")
+                                                       .append(blockMeta);
                                     }
                                 }
                             }
                             if (j != ConfigGeneralSettings.generalFlatWorldConfig.length - 1) {
-                                if (!flatworldconfig.endsWith(";")) {
-                                    flatworldconfig += ",";
+                                if (!flatworldconfig.toString().endsWith(";")) {
+                                    flatworldconfig.append(",");
                                 }
                             }
                         }
-                        this.field_146334_a = flatworldconfig;
+                        //Set preset
+                        this.field_146334_a = flatworldconfig.toString();
                     }
                 }
             }
+            //Update GUI
+            field_146331_K.setInt(this, worldGenType);
 
-
-            field_146331_K.setInt(this, WorldGenerator);
-
-            Field field_146344_y = GuiCreateWorld.class.getDeclaredField("field_146344_y");
-            field_146344_y.setAccessible(true);
-
-
-            func_146319_h.invoke(this);
```

Comment:
Any reason for removing this invocation? It's named `updateDisplayState` in 1.12 and I guess it's still needed?

Reply:
Oh nvm, `super#initGui` already does that